api/filters/checkBrokerCredentials.go

```diff
@@ -25,24 +27,29 @@ func (*CheckBrokerCredentialsFilter) Name() string {
 
 func (*CheckBrokerCredentialsFilter) Run(req *web.Request, next web.Handler) (*web.Response, error) {
 	brokerUrl := gjson.GetBytes(req.Body, "broker_url")
```

Comment:
it happens in api.go before saving the object, validate function of the object is called 
// validate validates the specified value in case it implements InputValidator
func validate(value interface{}) error {
	if input, ok := value.(InputValidator); ok {
		if err := input.Validate(); err != nil {
			return &HTTPError{
				ErrorType:   "BadRequest",
				Description: err.Error(),
				StatusCode:  http.StatusBadRequest,
			}
		}
	}
	return nil
}